libcontainer/cgroups/systemd/v2.go

```diff
@@ -140,16 +145,20 @@ func (m *unifiedManager) Apply(pid int) error {
 	properties = append(properties, resourcesProperties...)
 	properties = append(properties, c.SystemdProps...)
 
-	if err := startUnit(unitName, properties); err != nil {
+	dbusConnection, err := getDbusConnection(m.rootless)
+	if err != nil {
 		return err
 	}
+	if err := startUnit(dbusConnection, unitName, properties); err != nil {
+		return errors.Wrapf(err, "error while starting unit %q with properties %+v", unitName, properties)
+	}
 
 	_, err = m.GetUnifiedPath()
 	if err != nil {
 		return err
 	}
 	if err := fs2.CreateCgroupPath(m.path, m.cgroups); err != nil {
-		return err
+		return errors.Wrapf(err, "error while creating directory %q", m.path)
```

Comment:
In many cases (except one) you'll end up printing the directory name twice here.

It's better to modify `fs2.CreateCgroupPath` to make sure path is always present in the error message (need to wrap error returned from `neededControllers` in there), and don't do any wrapping here.

Reply:
updated